### PR TITLE
settings. Show a warning message if user edit hostname

### PR DESCRIPTION
**Pull request by tonyco97**
https://github.com/nethesis/ns8-nethvoice/issues/211#issuecomment-2139164921

**If the user starts to edit the hostname field, a warning message will be shown.**

image

### CODE REVIEW

**Comment by tonyco97** [comment]
Fixed also missing translations: https://github.com/nethesis/ns8-nethvoice/issues/212

**Review by Amygos — APPROVED** [review]

**Comment by tonyco97** [comment]
**Update warning message and custom NS Dropdown warning message**

image

**Review by edospadoni — APPROVED** [review]

File: ui/public/i18n/en/translation.json [hunk 1]
```diff
@@ -51,7 +52,10 @@
     "timezone_tooltip": "Set the default timezone for current NethVoice installation",
     "create_nethvoice_adm": "Create NethVoice Administrator for users management",
     "remove_nethvoice_adm": "Remove NethVoice Administrator for users management",
-    "set_nethvoice_adm_password": "Set NethVoice Administrator password for users management"
+    "set_nethvoice_adm_password": "Set NethVoice Administrator password for users management",
+    "Error message hostname": "By changing this field all user configurations will be reset.",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:56]: If the string "Error message hostname" is not returned from the backend, let's use something like "error_message_hostname"

File: ui/public/i18n/en/translation.json [hunk 2]
```diff
@@ -51,7 +52,10 @@
...
+    "nethvoice_host_format": "The format of the hostname is not valid. It must be a valid domain name.",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:57]:
```suggestion
    "nethvoice_host_format": "Must be a valid domain name",
```

File: ui/public/i18n/en/translation.json [hunk 3]
```diff
@@ -51,7 +52,10 @@
...
+    "nethcti_ui_host_format": "The format of the hostname is not valid. It must be a valid domain name."
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:58]:
```suggestion
    "nethcti_ui_host_format": "Must be a valid domain name."
```

File: ui/public/i18n/it/translation.json [hunk 4]
```diff
@@ -51,7 +51,10 @@
     "timezone_tooltip": "Imposta la timezone di default per questa installazione di NethVoice",
     "create_nethvoice_adm": "Creazione NethVoice Administrator per gestione utenti",
     "remove_nethvoice_adm": "Rimozione NethVoice Administrator per gestione utenti",
-    "set_nethvoice_adm_password": "Impostazione NethVoice Administrator password per gestione utenti"
+    "set_nethvoice_adm_password": "Impostazione NethVoice Administrator password per gestione utenti",
```
Comment (andre8244) [inline ui/public/i18n/it/translation.json:54]: Non-english translations should be done on Weblate, not editing the translations file directly

File: ui/src/views/Settings.vue [hunk 5]
```diff
@@ -97,6 +98,7 @@
               :disabled="loadingState"
               :invalid-message="error.nethvoice_admin_password"
               ref="nethvoice_admin_password"
+              :type="passwordFieldType"
```
Comment (andre8244) [inline ui/src/views/Settings.vue:101]:
```suggestion
              type="password"
```
`passwordFieldType` has always value `password`, no need to use a variable

File: ui/src/views/Settings.vue [hunk 6]
```diff
@@ -187,8 +191,24 @@ export default {
...
         timezone: "",
       },
+      warning: {
+        user_domain: "",
+      },
     };
   },
+  mounted() {
+    this.previousUserDomain = this.form.user_domain;
+    this.$refs.user_domain.$el.addEventListener(
```
Comment (andre8244) [inline ui/src/views/Settings.vue:201]: Vue support events like `@input` for this kind of things, see https://vuejs.org/guide/essentials/forms.html#form-input-bindings

**Pull request merged**